tests/test_basic_ops_dict.py

```diff
@@ -398,10 +403,29 @@ def test_dict_pop_error(cfg: Dict[Any, Any], key: Any, expectation: Any) -> None
             "incompatible_key_type",
             False,
         ),
+        (  # unexpected failure?
+            DictConfig(content={Enum1.FOO: "foo"}, key_type=Enum1, element_type=str),
+            "FOO",
+            False,
+        ),
+        # int key type
         ({1: "a", 2: {}}, 1, True),
         ({1: "a", 2: {}}, 2, True),
         ({1: "a", 2: {}}, 3, False),
         ({1: "a", 2: "???"}, 2, False),
+        # float key type
+        ({1.1: "a", 2.2: {}}, 1.1, True),
+        ({1.1: "a", 2.2: {}}, "1.1", False),
+        ({1.1: "a", 2.2: {}}, 2.2, True),
+        ({1.1: "a", 2.2: {}}, 3.3, False),
+        ({1.1: "a", 2.2: "???"}, 2.2, False),
+        # bool key type
+        ({True: "a", False: {}}, True, True),
+        ({True: "a", False: {}}, False, True),
+        ({True: "a", False: {}}, "no", False),
+        ({True: "a", False: {}}, 1, False),  # unexpected failure
+        ({True: "a", False: {}}, None, False),  # unexpected AssertionError
```

Comment:
I was surprised that these tests fail too. The first test fails because
```
1 in OmegaConf.create({True: "a", False: {}})
```
is `True` (which suggests that `1` is being coerced to `bool`), and the second test fails because `assert ret is not None` triggers `AssertionError` in [`DictConfig._s_validate_and_normalize_key`](https://github.com/7iW/omegaconf/blob/ce97297d6046e97bd22531d90d7357cda2b021fb/omegaconf/dictconfig.py#L305).

Reply:
It seems that `(1 in {True: "a"}) is True` is standard behavior in python, so the first test result above is as expected.

Reply:
The second test failure above is resolved in https://github.com/omry/omegaconf/pull/484/commits/61c08fdf713a9811519814da943eaa4b3534148f by changing
```
`assert ret is not None`
```
to
```
if ret is None: raise ValidationError
```